x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/install_prebuilt_rules/review_installation.ts

```diff
@@ -0,0 +1,844 @@
+          expect(
+            await reviewPrebuiltRulesToInstall(
+              supertest,
+              {
+                page: -1,
+              },
+              400
+            )
+          ).toMatchObject({
+            message: '[request body]: page: Number must be greater than or equal to 1',
+          });
+        });
+
+        it('rejects invalid "per_page" parameter', async () => {
+          expect(
+            await reviewPrebuiltRulesToInstall(
+              supertest,
+              {
+                per_page: -1,
+              },
+              400
+            )
+          ).toMatchObject({
+            message: '[request body]: per_page: Number must be greater than or equal to 1',
+          });
+
+          expect(
+            await reviewPrebuiltRulesToInstall(
+              supertest,
+              {
+                per_page: 10_001,
+              },
+              400
+            )
+          ).toMatchObject({
+            message: '[request body]: per_page: Number must be less than or equal to 10000',
+          });
+        });
+      });
+    });
+
+    describe('Tags', () => {
+      it('returns tags from all installable rules even if a filter is provided', async () => {
+        const ruleAssets = [
+          createRuleAssetSavedObject({ rule_id: 'rule-1', version: 1, tags: ['tag-a', 'tag-b'] }),
+          createRuleAssetSavedObject({ rule_id: 'rule-2', version: 1, tags: ['tag-b'] }),
+          createRuleAssetSavedObject({ rule_id: 'rule-3', version: 1, tags: ['tag-c'] }),
+        ];
+
+        await createPrebuiltRuleAssetSavedObjects(es, ruleAssets);
+
+        const response = await reviewPrebuiltRulesToInstall(supertest, {
+          filter: { fields: { tags: { include: ['tag-a'] } } },
+        });
+
+        expect(response.stats.tags).toEqual(['tag-a', 'tag-b', 'tag-c']);
+      });
+
+      it('tags are sorted alphabetically', async () => {
+        const ruleAssets = [
+          createRuleAssetSavedObject({
+            rule_id: 'rule-1',
+            version: 1,
+            tags: ['tag-b', 'tag-a', 'tag-c'],
+          }),
+        ];
+
+        await createPrebuiltRuleAssetSavedObjects(es, ruleAssets);
+
+        const response = await reviewPrebuiltRulesToInstall(supertest);
+
+        expect(response.stats.tags).toEqual(['tag-a', 'tag-b', 'tag-c']);
+      });
+    });
+
+    describe('Sorting', () => {
+      describe('by rule name', () => {
+        let ruleAssets = [];
+
+        beforeEach(async () => {
+          ruleAssets = [
+            createRuleAssetSavedObject({ rule_id: 'rule-1', version: 1, name: 'Rule 1' }),
+            createRuleAssetSavedObject({ rule_id: 'rule-3', version: 1, name: 'Rule 3' }),
+            createRuleAssetSavedObject({
+              rule_id: 'rule-2',
+              version: 1,
+              name: 'Rule 2, previous version',
+            }),
+            createRuleAssetSavedObject({ rule_id: 'rule-2', version: 2, name: 'Rule 2' }),
+          ];
```

Comment:
It's better to have the setup logic in the each test. We'll have duplications this way but it will help avoid jumping when reading the tests. On of that you may use test scenario focused data in each case.